tests/unit/test_database_migration.py

```diff
@@ -0,0 +1,131 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Unit tests for Flask charm database integration."""
+
+import ops
+import pytest
+from ops.testing import Harness
+
+from charm_state import CharmState
+from constants import FLASK_CONTAINER_NAME
+from database_migration import DatabaseMigration
+from exceptions import CharmConfigInvalidError
+from flask_app import FlaskApp
+from webserver import GunicornWebserver
+
+
+def test_database_migration(harness: Harness):
+    """
+    arrange: none
```

Comment:
Please, could you give more details about this test? 
I see that you set Flaskapp and then restart Flask three times always checking the database_migration_history but I think I didn't follow why there is no "database-migration-2.sh" in the last check.